### PR TITLE
Improve the exceptions that are thrown by Kotlin-Guiced

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Before:
```
com.google.inject.CreationException: Unable to create injector, see the following errors:

1) A binding to org.jlleitschuh.guice.Interface was already configured at org.jlleitschuh.guice.BinderScope.bind(BinderScope.kt:43).
  at org.jlleitschuh.guice.BinderScope.bind(BinderScope.kt:43)

```
Now:
```
com.google.inject.CreationException: Unable to create injector, see the following errors:

1) A binding to org.jlleitschuh.guice.Interface was already configured at org.jlleitschuh.guice.ModuleTest$module binding exceptions provide meaningful exceptions$module$1.invoke(ModuleTest.kt:100).
  at org.jlleitschuh.guice.ModuleTest$module binding exceptions provide meaningful exceptions$module$1.invoke(ModuleTest.kt:104)
```

### CODE REVIEW

File: kotlin-guiced.gradle.kts [hunk 1]
```diff
@@ -69,10 +70,26 @@ subprojects {
...
         "testRuntime"(junitJupiter("junit-jupiter-engine"))
         "testRuntime"(create(group = "org.junit.platform", name = "junit-platform-launcher", version = "1.3.1"))
     }
 
+    tasks.withType<KotlinCompile>().configureEach {
+        // Kotlin incremental compilation is enabled by default
+        kotlinOptions {
+            jvmTarget = "1.8"
+
+            freeCompilerArgs += "-Xprogressive"
```
Comment (Octogonapus) [inline kotlin-guiced.gradle.kts:83]: This should be `-progressive`.

Comment (JLLeitschuh) [reply kotlin-guiced.gradle.kts:83]: I knew it would be good to have you on this PR. Thanks!

Comment (Octogonapus) [reply kotlin-guiced.gradle.kts:83]: `-Xprogressive` still works, though. The original 1.2.51 (IIRC) release notes said to use `-Xprogressive` but the Kotlin docs say `-progressive`. The compiler actually looks for `-Xprogressive` and will print out a warning to you saying to use `-progressive` instead.

**Review by Octogonapus — APPROVED** [review]

File: kotlin-guiced.gradle.kts (hunk 1 above)
Comment (JLLeitschuh) [reply kotlin-guiced.gradle.kts:83]: We aren't at Kotlin 1.3 for this project yet so this can't be done yet, unfortunately.

**Pull request merged**